pkg/minishift/profile/profile.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minishift/config"
+)
+
+//Returns the machine name after taking in to account if --profile or
+//profile information available in allinstance config
+func GetProfileName() string {
```

Comment:
I pass the path to the config and the profile name into the function. This applies to everything. Your methods are tiring themselves to these contacts, making things hard to test as well as change. 

If you pass parameters, you can easily write tests creating arbitrary test files or even creating them in memory. Right now this is completely tied to the constants.